dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/kafka/flatmap/KafkaFlatMapServlet.java

```diff
@@ -0,0 +1,58 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.reactive.messaging.fat.kafka.flatmap;
+
+import static java.time.temporal.ChronoUnit.SECONDS;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+
+import java.time.Duration;
+import java.util.List;
+
+import javax.servlet.annotation.WebServlet;
+
+import org.junit.Test;
+
+import com.ibm.ws.microprofile.reactive.messaging.fat.kafka.framework.AbstractKafkaTestServlet;
+import com.ibm.ws.microprofile.reactive.messaging.fat.kafka.framework.SimpleKafkaReader;
+import com.ibm.ws.microprofile.reactive.messaging.fat.kafka.framework.SimpleKafkaWriter;
+
+/**
+ *
+ */
+@WebServlet("/flatMapTest")
+public class KafkaFlatMapServlet extends AbstractKafkaTestServlet {
+
+    /**  */
+    private static final long serialVersionUID = 1L;
+
+    public static final String IN_TOPIC = "flat-map-in";
+    public static final String OUT_TOPIC = "flat-map-out";
+    public static final String APP_GROUPID = "KafkaFlatMapTest-group";
+
+    @Test
+    public void testFlatMap() throws InterruptedException {
+        SimpleKafkaWriter<String> writer = kafkaTestClient.writerFor(IN_TOPIC);
+        SimpleKafkaReader<String> reader = kafkaTestClient.readerFor(OUT_TOPIC);
+
+        long offset = kafkaTestClient.getTopicOffset(IN_TOPIC, APP_GROUPID);
+
+        writer.sendMessage("abc");
+        writer.sendMessage("abcd");
+        writer.sendMessage("abcde");
+        writer.sendMessage("abcdef");
+
+        List<String> messages = reader.waitForMessages(2, Duration.of(10, SECONDS));
```

Comment:
can you use a constant for the timeouts?

Reply:
Yes